### PR TITLE
total order seek

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
When use prefix extractor, prefix model seek is used by default, if we want iterator all keys, we should set `total_order_seek` to true.
@siddontang @BusyJay @disksing PTAL

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
ping

**Comment by siddontang** [comment]
I think we should add more tests after #63 merged.

**Comment by siddontang** [comment]
We can add more test now. @zhangjinpeng1987

**Comment by zhangjinpeng1987** [comment]
ping

**Comment by zhangjinpeng1987** [comment]
@siddontang @BusyJay PTAL

File: test/test_iterator.rs [hunk 1]
```diff
@@ -186,3 +200,53 @@ fn read_with_upper_bound() {
...
+
+        db.put_opt(b"k2-0", b"a", &wopts).unwrap();
+        db.put_opt(b"k2-1", b"b", &wopts).unwrap();
+        db.put_opt(b"k2-2", b"c", &wopts).unwrap();
+        db.flush(true).unwrap();
+
+        let mut iter = db.iter();
+        iter.seek(SeekKey::Key(b"k1-0"));
+        let mut key_count = 0;
+        while iter.valid() {
```
Comment (siddontang) [inline test/test_iterator.rs]: maybe we can check key value here too.

File: test/test_iterator.rs [hunk 2]
```diff
@@ -186,3 +200,53 @@ fn read_with_upper_bound() {
...
+        let mut iter = db.iter_opt(ropts);
+        iter.seek(SeekKey::Key(b"k1-0"));
+        let mut key_count = 0;
+        while iter.valid() {
+            // iterator all sst files
+            key_count = key_count + 1;
+            iter.next();
+        }
+        assert!(key_count == 6);
+    }
```
Comment (siddontang) [inline test/test_iterator.rs]: should add a test for memtable prefix bloom filter + total order seek

File: test/test_iterator.rs [hunk 3]
```diff
@@ -213,25 +213,38 @@ fn test_total_order_seek() {
     opts.set_prefix_extractor("FixedSuffixTransform",
                               Box::new(FixedSuffixTransform { suffix_len: 2 }))
         .unwrap();
+    // also create prefix bloom for memtable
+    opts.set_memtable_prefix_bloom_size_ratio(0.1 as f64);
 
     {
```
Comment (siddontang) [inline test/test_iterator.rs]: can remove the `{`

**Pull request merged**